django/contrib/admin/static/admin/js/inlines.js

```diff
@@ -140,7 +140,25 @@
             if ((maxForms.val() === '') || (maxForms.val() - forms.length) > 0) {
                 addButton.parent().show();
             }
-            // Hide the remove buttons if at min_num.
+
+            const isFormNotEnough = function() {
+                const totalFormNum = ~~totalForms.val();
+                const initialFormNum = ~~initialForms.val();
+                const minFormNum = ~~minForms.val();
+                const extraFormNum = ~~extraForms.val();
+
+                const emptyFormNum = totalFormNum - initialFormNum;
+                if (initialFormNum >= emptyFormNum) {
+                    return emptyFormNum < extraFormNum;
+                }
+                return emptyFormNum < extraFormNum + minFormNum - initialFormNum;
+            }
+            // Auto add form when forms are not enough
+            if (isFormNotEnough()) {
+                $('.' + options.addCssClass +' a').click();
+            }
+
```

Comment:
Sorry, why delete the logic here? The failure of test is probably related to it. 

(I found I have no permission to push code to the pr, I couldn't chang anything.